src/librbd/journal/Types.h

```diff
@@ -194,8 +199,10 @@ struct SnapRenameEvent : public SnapEventBase {
   SnapRenameEvent(uint64_t op_tid, uint64_t src_snap_id,
                   const std::string &src_snap_name,
                   const std::string &dest_snap_name)
-    : SnapEventBase(op_tid, dest_snap_name), snap_id(src_snap_id),
-      src_snap_name(src_snap_name) {
+    : SnapEventBase(op_tid, cls::rbd::UserSnapshotNamespace(),
```

Comment:
Since ```SnapRenameEvent``` encodes a snap id and source snap name, it can no longer inherit from ```SnapEventBase``` since ```SnapEventBase``` encode function will inject new parameters into the struct between ```SnaRenameEvent```'s parameters.